ExpressApp/controllers/api-articles-controller.js

```diff
@@ -13,3 +18,20 @@ exports.getArticles = (req, res) => {
     res.status(200).send({ articles });
   });
 };
+
+exports.getArticleComments = (req, res, next) => {
+  const articlePromises = [getAllComments(req.params.article_id)];
+
+  if (req.params.article_id) {
+    articlePromises.push(
+      checkExists("articles", "article_id", req.params.article_id)
+    );
+  }
+
```

Comment:
You could include your checkExist within your Promise.all and it should reject if no article is found and continue if all is good.